src/dataframe/dataframe.jl

```diff
@@ -1202,8 +1241,8 @@ end
 
 # hcat! for 2 arguments, only a vector or a data frame is allowed
 function hcat!(df1::DataFrame, df2::AbstractDataFrame;
-               makeunique::Bool=false, copycols::Bool=true)
-    u = add_names(index(df1), index(df2), makeunique=makeunique)
+               makeunique::Bool=false, mergeduplicates=nothing, copycols::Bool=true)
+    u = add_names(index(df1), index(df2), makeunique=true, mergeduplicates=mergeduplicates)
```

Comment:
this implementation looks incorrect. Where do you use the `mergeduplicates` if passed?

Reply:
It's presence is necessary to prevent the error throw in `add_names`.  

That said, I don't see any use of `merge` or `merge!` in the codebase, so the change was for completeness, since `add_names` gets used in `hcat!` and needs to be sensitive to `mergeduplicates` there.

Reply:
But I do not understand why. `add_names` works on `index` so it cannot take into account `mergeduplicates` anyway. Why and how do you think it should?

Reply:
To be clear - I saw implementation of `add_name` that you have changed. I just feel that `add_names` should not handle the `mergeduplicates` kwarg because it makes the design non-obvious.

Reply:
Maybe to further clarify my thinking. We need to make sure that the design we use is simple. The reason is that this package is large and it is important to make sure that future developers have an easy way to understand how everything works. That is the reason I recommend taking "small steps", as this ensures that these small steps have clean design.